File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string prefix = "";
        int n=strs.size();
        
        for (int i=0; i<strs[0].length(); ++i) {
            // since vector is sorted, compare each letter in first and last string
            if (strs[0][i] != strs[n-1][i]) 
                return prefix;
            prefix += strs[0][i];
            
        }
        return prefix;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        n = len(strs)-1
        prefix = ""
        
        for i in range(len(strs[0])):
            if strs[0][i] == strs[n][i]: # comparing first and last strings, because list is sorted
                prefix += strs[0][i]
            else:
                break
            
        return prefix